base/broadcast.jl

```diff
@@ -316,8 +316,7 @@ function .//(A::AbstractArray, B::AbstractArray)
     broadcast!(//, Array(type_rdiv(eltype(A), eltype(B)), broadcast_shape(A, B)), A, B)
 end
 
-type_pow(T,S) = promote_type(T,S)
-type_pow{S<:Integer}(::Type{Bool},::Type{S}) = Bool
+type_pow(T,S) = promote_type(Base.return_types(^,(T,S))...)
```

Comment:
Here's the reason i didn't use `typeof(one(T)^one(S))`:

```
julia> Real[2.1].^[2]
ERROR: InexactError()
 in setindex! at array.jl:328
 in _F_ at broadcast.jl:96
 in broadcast! at broadcast.jl:227
 in .^ at broadcast.jl:326
```

because `one(Real) === Int`.